### ISSUE TITLE
When will we use static / factory methods? 

### ISSUE THREAD

**Issue by rena-ele** [nus-cs2030/2122-s2#50]

**Comment by SpectralMusket** [comment]
So far I've only used it to transform data to fit into a super(x,y,z,etc) constructor inside a child class.
For some reason I don't know, Java only allows you to put super() constructors in the first line of the child constructor.
So you cant manipulate the parameters and then pass them into the super() constructor, you have to transform them inside the brackets.

For example in Lab 3 the cruises example, since the children cruises are practically the same as a normal Cruise but their parameters are handled differently (eg. takes 2x longer to load or sth, I don't remember), their constructor is basically calling super(x,y,z) but with x, y and z modified.

So instead of writing super(x/60, y*2, Math.power(z, 4)) etc. in 1 long line you can make a static method to transform the parameters into what you want, like super(bigLoadTime(x), bigArrivalTime(y), bigWhatever(z)).

I'm sure there are other uses as well like the String.format() method, where the method is often used to manipulate data that is not part of an instance (? not sure whether im using the right word here), but I'm not sure how to describe and identify those situations.

**Comment by ReganKvioke** [comment]
I only know 2 reasons why we use static for CS2030.
1. To create a variable for the magic number, so like "private static final HOURS_TO_MIN = 60;"
2. In the child class, you need to do some computation in the constructor, so you have a "private static method" that allows you to use this method in the super().

**Comment by shotnothing** [comment]
I don't think this course expects us to do this, but static properties/methods are very often used to access data/functions that does not make sense to create an object for. 

For example, notice when you use `Math.PI` you dont first do `Math math = new Math()`. That is because `PI `is a static property in `Math`.

Similarly when you call `System.out.println()`, you do not create an instance of `System`, as that would not make sense.
